### PR TITLE
Protect against invalid column configs

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Prereq: https://github.com/CrowdStrike/ember-headless-table/pull/82, #83, #84
Resolves: https://github.com/CrowdStrike/ember-headless-table/issues/70

See changeset + test for details.

tl;dr: we can improve DX by failing as early as we can when accessing the column configs for the first time.

Without this, we'd get very hard to debug errors (especially around column re-ordering where the key of a column is very important)

### CODE REVIEW

File: ember-headless-table/src/-private/table.ts [hunk 1]
```diff
@@ -52,24 +53,29 @@ export class Table<DataType = unknown> extends Resource<Signature<DataType>> {
 
   /**
    * @private
+   *
+   * Unused for now, may be used in the future.
+   * This data is collected along with the scrollContainerWidth,
```
Comment (nicolechung) [inline ember-headless-table/src/-private/table.ts:58]:
```suggestion
   * This data is collected along with the scrollContainerWidth, (which is currently in use)
```

File: .changeset/slow-eagles-switch.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+'ember-headless-table': patch
+---
+
+Prevent hard-to-debug issues that occur with incorrect column configs.
+One such way problems can occur is when the `key` property is duplicated
+for multiple column configs. 
+
+This is now eagerly prevented via dev-time Error.
```
Comment (nicolechung) [inline .changeset/slow-eagles-switch.md:9]:
```suggestion
This is now eagerly prevented via dev-time error when tests are run
```

rough suggestion tho

**Review by ynotdraw — APPROVED** [review]
Awesome - this is a great addition!

File: .changeset/slow-eagles-switch.md (hunk 2 above)
Comment (NullVoxPopuli) [reply .changeset/slow-eagles-switch.md:9]: the error will still occur when a consumer's app is in development :thinking: 
Is there another way I can make this more clear?

File: ember-headless-table/src/-private/table.ts (hunk 1 above)
Comment (NullVoxPopuli) [reply ember-headless-table/src/-private/table.ts:58]: thanks! this reads much clearer!

File: .changeset/slow-eagles-switch.md (hunk 2 above)
Comment (nicolechung) [reply .changeset/slow-eagles-switch.md:9]: No this is me totally misreading what I read.

**Pull request merged**